crates/supervisor/storage/src/chaindb_factory.rs

```diff
@@ -98,7 +104,16 @@ impl FinalizedL1Storage for ChainDbFactory {
         }
         *guard = Some(block);
 
-        // todo: update safety head ref for finalized level for all chains
+        // update all chain databases finalized safety head refs
+        let dbs = self.dbs.read().map_err(|err| {
+            error!(target: "supervisor_storage", %err, "Failed to acquire read lock on databases");
+            StorageError::LockPoisoned
+        })?;
+        for (chain_id, db) in dbs.iter() {
+            if let Err(err) = db.update_finalized_head_ref(block) {
+                error!(target: "supervisor_storage", chain_id = *chain_id, %err, "Failed to update finalized L1 in chain database");
```

Comment:
```suggestion
                error!(target: "supervisor_storage", chain_id, %err, "Failed to update finalized L1 in chain database");
```
pretty sure `u64` and `&u64` display format the same, if anything i would imagine you need to add `%chain_id` to specify display formatting